src/scripts/genXplatLttng.py

```diff
@@ -239,158 +267,193 @@ def generateLttngHeader(providerName,allTemplates,eventNodes):
+    
+    for paramName in fnSig.paramlist:
+        fnparam     = fnSig.getParam(paramName)
+        paramname   = fnparam.name
+
+        if fnparam.winType == "win:UnicodeString":
+            result.append("    INT " + paramname + "_path_size = -1;\n")
+            result.append("    PathCharString " + paramname + "_PS;\n")
+            result.append("    INT " + paramname + "_full_name_path_size")
+            result.append(" = (PAL_wcslen(" + paramname + ") + 1)*sizeof(WCHAR);\n")
+            result.append("    CHAR* " + paramname + "_full_name = ")
+            result.append(paramname + "_PS.OpenStringBuffer(" + paramname + "_full_name_path_size );\n")
+            result.append("    if (" + paramname + "_full_name == NULL )")
+            result.append("    { return ERROR_WRITE_FAULT; }\n")
+
+    result.append("\n")
+
+    #emit tracepoints
+    fnSig   = template.signature
+        
+    if not shouldPackTemplate(template):
+        linefnbody = ["    do_tracepoint(%s,\n        %s" % (providerName, eventName)]
+
+        for params in fnSig.paramlist:
+            fnparam     = fnSig.getParam(params)
+            wintypeName = fnparam.winType
+            winCount    = fnparam.count
+            paramname   = fnparam.name
+            ctf_type    = ctfDataTypeMapping.get(winCount)
+
+            line = "        "
+            if not ctf_type:
+                ctf_type    = ctfDataTypeMapping[wintypeName]
+
+            if ctf_type == "ctf_string" and wintypeName == "win:UnicodeString":
+                #emit code to convert unicode to ansi string
+
+                result.append("    " + paramname+ "_path_size = WideCharToMultiByte( CP_ACP, 0, ")
+                result.append(paramname + ", -1, ")
+                result.append(paramname + "_full_name, ")
+                result.append(paramname + "_full_name_path_size, NULL, NULL );\n")
+
+                result.append("    _ASSERTE(" +paramname+ "_path_size < " )
+                result.append(paramname + "_full_name_path_size );\n    ")
+
+                result.append(paramname + "_PS.CloseBuffer(" + paramname + "_path_size );\n")
+                result.append("    if( " + paramname + "_path_size == 0 ){ return ERROR_INVALID_PARAMETER; }\n")
+
+                line += paramname + "_full_name"
+                linefnbody.append(line)
+                continue
+
+            elif ctf_type == "ctf_sequence" or wintypeName == "win:Pointer":
+                line += "(" + lttngDataTypeMapping[wintypeName]
+                if not lttngDataTypeMapping[winCount] == " ":
+                    line += lttngDataTypeMapping[winCount]
+
+                line += ") "
+                linefnbody.append(line + paramname)
+
+            else:
+                linefnbody.append(line + paramname)
+
+        linefnbody = ",\n".join(linefnbody) + ");\n"
+        result.append(linefnbody)
+        return ''.join(result)
+
+    else:
+        header = """
+    char stackBuffer[%s];
```

Comment:
I think all required machinery is present in PathCharString, from the looks of your custom code takes char* as inputs, so it should be easy to adapt it to PathCharString, It will cleanup the code and reuse the logic which will be maintained well